server/db/models/conversation.js

```diff
@@ -23,16 +35,73 @@ Conversation.findConversationById = async function (conversationId) {
   return conversation;
 }
 
-Conversation.createConversation = async function (senderId, recipientId) {
+Conversation.createConversation = async function (senderId, recipientId ) {
   if (isNaN(senderId) || isNaN(recipientId)) {
     throw new Error("Error: Conversation could not be created. Two valid user Id's are needed.")
   }
   conversation = await Conversation.create({
     user1Id: senderId,
     user2Id: recipientId,
+    lastMessageReadUser1: null,
+    lastMessageReadUser2: null,
+    unseenCount: 0,
```

Comment:
If we know the last read message of each user, `unseenCount` can be calculated. We do not need to persist this value in database.

Reply:
Excellent, I will investigate those issues and fix them before merging. That's a great insight on using the last read message to calculate the unseenCount value. Thank you!